rust/src/datafusion/logical_expr.rs

```diff
@@ -0,0 +1,109 @@
+// Copyright 2023 Lance Developers.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Extends logical expression.
+
+use arrow_schema::DataType;
+use datafusion::logical_expr::Operator;
+use datafusion::scalar::ScalarValue;
+use datafusion::{logical_expr::BinaryExpr, prelude::*};
+
+use crate::datatypes::Schema;
+use crate::{Error, Result};
+
+/// Resolve a Value
+fn resolve_value(expr: &Expr, data_type: &DataType) -> Result<Expr> {
```

Comment:
needs to support list value for `item in (...)` ?

Reply:
Ok, can do.